src/actors/containerization/services/apache/apache_generator/apache_generator.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+
+"""
+Author: tmeszaro@redhat.com
+
+
+Apache generator actor. Uses data from apache_scanner, creates image with buildah
+and pushes this image into the specified registrty.
+
+
+Running the container:
+
+- after pulling image from registry, run:
+
+$ docker run -dt -p HOST_PORT:CONTAINER_PORT IMAGE_ID
+
+
+Possible improvements:
+
+- build from this? registry.access.redhat.com/rhscl/httpd-24-rhel7
+- parse httpd -S or httpd -V to get main config location, so we can run httpd
+   with non-standard config location
+- copy also other things (like logs, etc.)
+- move image pushing into separate actor
+
+"""
+
+
+import json
+import os
+import sys
+import subprocess
+import shlex
+from distutils.spawn import find_executable
+
+import dnf
+
+
+def find_buildah():
+    def _is_valid(path):
+        return path and os.path.isfile(path) and os.access(path, os.X_OK)
+    path = os.path.abspath('buildah')
+    if not _is_valid(path):
+        path = find_executable('buildah')
+    if not _is_valid(path):
+        raise Exception('Could not find a valid buildah binary: {0}'.format(path))
+    return path
+
+BUILDAH = find_buildah()
+
+def execute(cmd):
+    out = subprocess.check_output(shlex.split(cmd))
+    return out.strip()
+
+def create_httpd_container():
+    cname = execute('{buildah} from fedora'.format(buildah=BUILDAH))
+    if not cname:
+        raise Exception('Error creating container')
+    return cname
+
+def copy_to_container(cname, src, dst):
+    if cname and src and dst:
+        execute('{buildah} copy {cname} {src} {dst}'.format(
+            buildah=BUILDAH,
+            cname=cname,
+            src=src,
+            dst=dst
+        ))
+
+def get_container_id(cname):
+    cntrs = json.loads(execute('{buildah} containers --json'.format(buildah=BUILDAH)))
+    ids = [x['id'].strip() for x in cntrs if x['containername'] == cname]
+    if len(ids) != 1:
+        raise Exception('Incorrect number of containers with name {0}'.format(cname))
+    return ids[0]
+
+if __name__ == "__main__":
+    INPUT = json.load(sys.stdin)
```

Comment:
why constant? I think you're using too much of them.
It's confusing.. i would use them for values that you know before execution, like:
```CONSTANT = "I know this value before exec and it can't change"```

Reply:
Constant because I also use constants for the variables that will not change the value during the execution.
I guess that is my habit from C/C++ and I agree that having tons of capitalized variables looks confusing.

Reply:
in that case you would have a lot of "constant" variables in Python, i suggest to change it :) 
since we don't have real constant in python